### ISSUE TITLE
Use ReadImage and WriteImage functions where appropriate

### ISSUE THREAD

**Issue by dzenanz** [InsightSoftwareConsortium/ITK#2802]
Setting up a reader or a writer just to use them once in a straightforward way requires 5 lines of code, and it can be accomplished by using the appropriate convenience function. 

`itk::ReadImage` and `itk::WriteImage` are available since ITK 5.2.0. See https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/pull/225
for an illustration of the changes required and https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/issues/226 for further discussion.

This can be done piece-wise. A possible starting point are [bundled examples](https://github.com/InsightSoftwareConsortium/ITK/tree/master/Examples) which are used in the [software guide](https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/). @josetascon did a good job in [ITKSphinxExamples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/commits?author=josetascon).

**Comment by sadhana-r** [comment]
I can work on this issue

**Comment by dzenanz** [comment]
Sadhana's effort should move us along, but will probably not fully resolve this. I therefore assigned this to "backlog" milestone.

**Comment by krupalbhat** [comment]
can i work on this issue?

**Comment by dzenanz** [comment]
Yes! #5163 was merged, the torch can pass to you now. Perhaps first consider https://github.com/InsightSoftwareConsortium/ITK/pull/5163#discussion_r1932895575?

Whatever path you take, I suggest first doing the changes on a smaller set of files (~10 or so), get that reviewed, and then apply it to more files.

**Comment by hjmjohnson** [comment]
Largely done. As of `main`, **41 of 45** files under `Examples/` use `itk::ReadImage` / `itk::WriteImage`; only the following 4 still use the legacy `ImageFileReader::New()` / `ImageFileWriter::New()` form:

- \`Examples/Filtering/DiffusionTensor3DReconstructionImageFilter.cxx\`
- \`Examples/Segmentation/LaplacianSegmentationLevelSetImageFilter.cxx\`
- \`Examples/RegistrationITKv4/DeformableRegistration1.cxx\`
- \`Examples/RegistrationITKv4/DeformableRegistration11.cxx\`

These four are non-trivial pipelines (multi-component DTI reader, registration with explicit deformation-field IO) where the convenience wrapper is less of a fit. Closing — any further per-file conversion can be tracked in a fresh, scoped issue if/when desired.
